clang/lib/Parse/ParseTemplate.cpp

```diff
@@ -1305,6 +1307,17 @@ ParsedTemplateArgument Parser::ParseTemplateTemplateArgument() {
     }
   }
 
+  // We do not allow to reference builtin templates that produce multiple
+  // values, they would not have a well-defined semantics outside template
+  // arguments.
+  if (auto *T = !Result.isInvalid()
+                    ? Result.getAsTemplate().get().getAsTemplateDecl()
+                    : nullptr;
+      T && isPackProducingBuiltinTemplate(T)) {
+    Actions.diagnoseMissingTemplateArguments(Result.getAsTemplate().get(),
+                                             Result.getLocation());
+  }
```

Comment:
I would prefer to either: do that check in Sema or introduce a new Parser diagnostic.
Calling Action.diagnoseMissingTemplateArguments for that does seem like a layering violation